docs/pages/database-access/guides/elastic.mdx

```diff
@@ -0,0 +1,293 @@
+
+</TabItem>
+</Tabs>
+
+## Step 2/5. Create a Teleport user
+
+(!docs/pages/includes/database-access/create-user.mdx!)
+
+## Step 3/5. Create a role mapping
+
+Define a role mapping in Elasticsearch to assign your Teleport user(s) or role(s) to an Elasticsearch role.
+The example below maps the Teleport user `alice` to the `user` role in Elasticsearch.
+
+```code
+$ curl -u elastic:your_elasticsearch_password -X POST "https://elasticsearch.example.com:9200/_security/role_mapping/mapping1?pretty" -H 'Content-Type: application/json' -d'
+{
+  "roles": [ "user"],
+  "enabled": true,
+  "rules": {
+    "field" : { "username" : "alice" }
+  },
+  "metadata" : {
+    "version" : 1
+  }
+}
+'
+```
+
+<Details title="Role Mapping with wildcards" opened={false}>
+
+In a scenario where Teleport is using [single sign-on](../../access-controls/sso.mdx) you may want to define a mapping for all users to a role:
+
+```code
+$ curl -u elastic:your_elasticsearch_password -X POST "https://elasticsearch.example.com:9200/_security/role_mapping/mapping1?pretty" -H 'Content-Type: application/json' -d'
+{
+  "roles": [ "monitoring"],
+  "enabled": true,
+  "rules": {
+    "field" : { "username" : "*@example.com" }
+  },
+  "metadata" : {
+    "version" : 1
+  }
+}
+'
+```
+
+</Details>
+
+## Step 4/5. Set up mutual TLS
+
+(!docs/pages/includes/database-access/tctl-auth-sign.mdx!)
+
+We will show you how to use `tctl auth sign` below.
+
+To securely access your Elasticsearch database, sign the certificate for the hostname Teleport will
+connect to.
+
+For example, if your Elasticsearch server is accessible at `elastic.example.com`,
+run:
+
+```code
+$ tctl auth sign --format=elastic --host=elastic.example.com --out=elastic --ttl=2160h
+Database credentials have been written to elastic.key, elastic.crt, elastic.cas.
+```
+
+(!docs/pages/includes/database-access/ttl-note.mdx!)
+
+The command will create three files:
+
+- `elastic.cas` with Teleport's certificate authority
+- `elastic.key` with a generated private key
+- `elastic.crt` with a generated host certificate
+
+You will need these files to enable mutual TLS on your Elasticsearch server.
+
+(!docs/pages/includes/database-access/rotation-note.mdx!)
+
+Use the generated secrets to enable mutual TLS in your `elasticsearch.yml` configuration
+file:
+
+```yml
+xpack.security.http.ssl:
+  certificate_authorities: /path/to/elastic.cas
+  certificate: /path/to/elastic.crt
+  key: /path/to/elastic.key
+  enabled: true
+  client_authentication: required
+  verification_mode: certificate
+
+xpack.security.authc.realms.pki.pki1:
+  order: 1
+  enabled: true
+  certificate_authorities: /path/to/elastic.cas
+```
+
+Once mutual TLS has been enabled, you will no longer be able to connect to the cluster without
+providing a valid client certificate. You can set `xpack.security.http.ssl.client_authentication`
+to `optional` to allow connections from clients that do not present a certificate.
```

Comment:
May be worth adding that without client certificate the user must authenticate using a username and password (unless a guest account is enabled, in which case they will have guest privileges).